quickwit/quickwit-storage/src/split.rs

```diff
@@ -59,11 +61,34 @@ async fn range_byte_stream_from_payloads(
         );
     }
 
-    let body = Body::wrap_stream(stream::iter(bytestreams).flatten());
-    let concat_stream = ByteStream::new(body.into());
+    let body = Body::wrap_stream(stream::iter(bytestreams).map(StreamAdaptor).flatten());
+    let concat_stream = ByteStream::new(SdkBody::from_body_0_4(body));
     Ok(concat_stream)
 }
 
+// With sdk 1.0, ByteStream no longer implement Stream, despite having analogous functions
+// this adaptor is just meant to make it implmeent Stream for places where we really need it
+#[pin_project]
+struct StreamAdaptor(#[pin] ByteStream);
+
+impl Stream for StreamAdaptor {
+    type Item = Result<Bytes, aws_smithy_types::byte_stream::error::Error>;
+
+    fn poll_next(
+        self: Pin<&mut Self>,
+        ctx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        self.project().0.poll_next(ctx)
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let (min_u64, max_u64) = self.0.size_hint();
+        let min = min_u64.try_into().unwrap_or(usize::MAX);
```

Comment:
```suggestion
        let min: usize = min_u64.try_into().unwrap_or(0);
```

Reply:
I think MAX is the more correct value. If we fail the conversion, min_u64 was larger than what an usize can handle on that platform. zero is an always valid value to return, but in this case MAX is also valid, and more informative

Reply:
oh I see! Yes you are right!

Reply:
Can you add a comment so future readers don't make the same mistake as me?